### PR TITLE
Added IAP codelab

### PR DESCRIPTION
**Pull request by renefloor**
Code for the first version of the in app purchases codelab.

### CODE REVIEW

File: in_app_purchases/complete/app/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# iap
```
Comment (domesticmouse) [inline in_app_purchases/complete/app/README.md:1]: In App Purchases codelab?

Comment (domesticmouse) [reply in_app_purchases/complete/app/README.md:1]: Alternatively, drop this readme if there isn't anything new here

**Review by domesticmouse — COMMENTED** [review]
Is it your intent to only have step_00 and complete, or are you going to break it down into a series of steps?

**Comment by domesticmouse** [comment]
@RedBrogdon any idea on how to fix the cla/google error? 

I don't see this error listed in https://opensource.google/docs/cla/#troubleshoot

**Comment by renefloor** [comment]
@googlebot I consent

**Comment by renefloor** [comment]
> Is it your intent to only have step_00 and complete, or are you going to break it down into a series of steps?

There will be a folder per step, but we're still changing some code, so we have to change that code in all the steps. Therefore we first want to complete the codelab before making all the folder steps. We can also use the creation of the steps to verify the codelab code

**Comment by BeMacized** [comment]
@googlebot I consent.

**Comment by domesticmouse** [comment]
@renefloor sounds good to me, thanks!

**Comment by domesticmouse** [comment]
Please add this codelab to the CI test set by updating `flutter_ci_script_stable.sh`, `flutter_ci_script_beta.sh` and `flutter_ci_script_dev.sh`.

**Comment by domesticmouse** [comment]
Looks like there are [some issues in the stable CI pipeline](https://github.com/flutter/codelabs/pull/98/checks?check_run_id=2471020720#step:7:733). You should be able to run the `flutter_ci_script_stable.sh` script locally to confirm you have fixed it.

```
== TESTING in_app_purchases/step_00/app
+ flutter analyze
Running "flutter pub get" in app...                              1,749ms
Analyzing app...                                                

   info • Unused import: 'package:dashclicker/constants.dart' • lib/logic/dash_purchases.dart:3:8 • unused_import
   info • Unused import: 'package:dashclicker/logic/firebase_notifier.dart' • lib/logic/dash_purchases.dart:7:8 • unused_import
   info • Private field could be final • lib/logic/dash_purchases.dart:29:8 • prefer_final_fields

3 issues found. (ran in 3.6s)
```

**Pull request merged**